ras/lshw.py

```diff
@@ -178,8 +178,15 @@ def test_prod_id_serial(self):
         else:
             product_path = '/proc/device-tree/model'
             serial_path = '/proc/device-tree/system-id'
-        product_name = genio.read_file(product_path).rstrip(' \t\r\n\0')
-        serial_num = genio.read_file(serial_path).rstrip(' \t\r\n\0')
+
+        if 'x86_64' in process.system_output('lscpu', shell='true').decode():
+            product_name = self.run_cmd_out("dmidecode | grep 'Product Name'\
+                    | head -1 | cut -d ':' -f 2")
+            serial_num = self.run_cmd_out("dmidecode | grep 'Serial Number'\
+                    | head -1 | cut -d ':' -f 2")
```

Comment:
@kalpanashtty It will be better if python parsing logic is used for the dmicdecode output, instead of shell piping to get the information

Reply:
Done python parsing instead of shell command.